.github/workflows/diff_report.yml

```diff
@@ -0,0 +1,150 @@
+        sed 's/Diff Regression config: //' temp > config
+        echo ::set-output name=config_link::$(cat config)
+        
+     - name: Set branch and head_label
+       id: branch
+       run: |
+        echo ::set-output name=user::$(cat user)
+        echo ::set-output name=ref::$(cat branch)
+      
+     - name: info
+       run: |
+        cat user
+        cat branch
+        echo ${{steps.branch.outputs.ref}}
+        echo ${{steps.branch.outputs.user}}:
+     
+  make_report:
+    runs-on: ubuntu-latest
+    needs: parse_body
+    if: needs.parse_body.outputs.proj_link != '' && needs.parse_body.outputs.config_link != ''
+    steps: 
+       
+     - name: Download files
+       run: |
+        wget -q ${{needs.parse_body.outputs.proj_link}} -O proj.properties
+        wget -q ${{needs.parse_body.outputs.config_link}} -O config.xml
+        
+     - name: Download checkstyle
+       uses: actions/checkout@v2
+       with:
+        repository: ${{needs.parse_body.outputs.user}}/checkstyle
+        ref: master
+        path: checkstyle
+        fetch-depth: 0
+        
+     - name: Check branches
+       run: |
+         cd checkstyle
+         git branch -r
+         cd ../
+     
+     - name: Download contribution
+       uses: actions/checkout@v2
+       with:
+        repository: checkstyle/contribution
+        path: contribution
+         
+     - name: Print branch
+       run: |
+         echo ${{needs.parse_body.outputs.branch}}
+         
+     - name: Relocate .xml and .properties 
+       run: |
+         mv config.xml ./contribution/checkstyle-tester/
+         mv proj.properties ./contribution/checkstyle-tester/
+         
+     - name: Generate report
+       run: |
+         bash
+         REF="origin/${{needs.parse_body.outputs.branch}}"
+         cd contribution/checkstyle-tester
+         sudo apt-get install mercurial
+         sudo apt install groovy
+         groovy diff.groovy -r ../../checkstyle -b master -p $REF -c config.xml -l proj.properties
+        
+     - name: Configure AWS Credentials
+       uses: aws-actions/configure-aws-credentials@v1
+       with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ env.AWS_REGION }}
+       
+     - name: Copy Report to AWS S3 Bucket
+       run: |
+        bash
+        TIME=`date +%Y%H%M%S`
+        FOLDER="${{needs.parse_body.outputs.branch}}_$TIME"
+        DIFF="./contribution/checkstyle-tester/reports/diff"
+        LINK="https://${{env.AWS_BUCKET_NAME}}.s3.${{env.AWS_REGION}}.amazonaws.com"
+        aws s3 cp $DIFF s3://${{env.AWS_BUCKET_NAME}}/$FOLDER/reports/diff/ --recursive
+        echo $LINK/$FOLDER/reports/diff/index.html > message
+        
+     - name: Set output
+       id: out
+       run: echo ::set-output name=message::$(cat message)   
+      
+     - name: Comment PR
+       uses:  kate2513/javascript-test-action@master 
```

Comment:
this is ok for now , but should be changed in next PRs

Reply:
created https://github.com/checkstyle/checkstyle/issues/8446